app/models/registration_system.rb

```diff
@@ -41,6 +43,29 @@ def self.verify_credentials_queue(options = {})
     MiqTask.generic_action_with_callback(task_opts, queue_opts)
   end
 
+  def self.update_rhsm_conf_queue(options = {})
+    options = options.clone
+    options[:password] = MiqPassword.try_encrypt(options[:password])
+    options[:registration_http_proxy_password] = MiqPassword.try_encrypt(options[:registration_http_proxy_password])
+
+    task_opts = {
+      :action => "Update Rhsm Config",
+      :userid => "system"
+    }
+
+    MiqRegion.my_region.miq_servers.each do |server|
+      queue_opts = {
+        :class_name  => "RegistrationSystem",
+        :method_name => "update_rhsm_conf",
+        :priority    => MiqQueue::HIGH_PRIORITY,
+        :args        => [options.delete_blanks],
```

Comment:
It looks like `options` isn't changing.
Could you move the `delete_blanks` up out of the loop?